src/data/sqlGenerator.ts

```diff
@@ -5,7 +5,7 @@ import otel from 'otel';
  * Generates a SQL string for the given QueryBuilderOptions
  */
 export const generateSql = (options: QueryBuilderOptions): string => {
-  const hasTraceIdFilter = options.meta?.isTraceIdMode && options.meta?.traceId
+  const hasTraceIdFilter = options.meta?.isTraceIdMode && options.meta?.traceId !== undefined;
```

Comment:
What if `traceId` is empty string?

Reply:
Good catch. I've reverted that change, my bad!